service/src/main/java/uk/gov/hmcts/reform/fpl/service/PlacementService.java

```diff
@@ -81,4 +85,23 @@ private Placement removeConfidentialDocuments(Element<Placement> placement) {
             .filter(placement -> placement.getValue().getChildId().equals(childId))
             .findFirst();
     }
+
+    public List<String> getBinaryUrlsForOrderAndNotices(List<Element<Placement>> placements,
+                                                        PlacementOrderAndNotices.PlacementOrderAndNoticesType type) {
+        return placements.stream()
+            .filter(element -> element.getValue().getOrderAndNotices() != null)
+            .map(element -> element.getValue().getOrderAndNotices())
+            .flatMap(Collection::stream)
+            .filter(element -> element.getValue().getType() == type)
+            .map(this::getBinaryUrl)
+            .filter(Strings::isNotEmpty)
+            .collect(toList());
+    }
+
+    private String getBinaryUrl(Element<PlacementOrderAndNotices> element) {
+        DocumentReference document = ofNullable(element.getValue().getDocument())
+            .orElse(DocumentReference.builder().build());
+
+        return document.getBinaryUrl();
+    }
```

Comment:
Is this really needed?

Reply:
This is more, if there is no document, better to be defensive imo